src/renderer/components/+workloads-overview/overview.tsx

```diff
@@ -17,81 +17,68 @@ import { cronJobStore } from "../+workloads-cronjobs/cronjob.store";
 import { Events } from "../+events";
 import { KubeObjectStore } from "../../kube-object.store";
 import { isAllowedResource } from "../../../common/rbac";
+import { namespaceStore } from "../+namespaces/namespace.store";
 
 interface Props extends RouteComponentProps<IWorkloadsOverviewRouteParams> {
 }
 
 @observer
 export class WorkloadsOverview extends React.Component<Props> {
+  @observable isLoading = false;
   @observable isUnmounting = false;
 
   async componentDidMount() {
-    const stores: KubeObjectStore[] = [];
-
-    if (isAllowedResource("pods")) {
-      stores.push(podsStore);
-    }
-
-    if (isAllowedResource("deployments")) {
-      stores.push(deploymentStore);
-    }
-
-    if (isAllowedResource("daemonsets")) {
-      stores.push(daemonSetStore);
-    }
-
-    if (isAllowedResource("statefulsets")) {
-      stores.push(statefulSetStore);
-    }
-
-    if (isAllowedResource("replicasets")) {
-      stores.push(replicaSetStore);
-    }
-
-    if (isAllowedResource("jobs")) {
-      stores.push(jobStore);
-    }
-
-    if (isAllowedResource("cronjobs")) {
-      stores.push(cronJobStore);
-    }
-
-    if (isAllowedResource("events")) {
-      stores.push(eventStore);
-    }
-
-    const unsubscribeList: Array<() => void> = [];
-
-    for (const store of stores) {
-      await store.loadAll();
-      unsubscribeList.push(store.subscribe());
-    }
+    const stores: KubeObjectStore[] = [
+      isAllowedResource("pods") && podsStore,
+      isAllowedResource("deployments") && deploymentStore,
+      isAllowedResource("daemonsets") && daemonSetStore,
+      isAllowedResource("statefulsets") && statefulSetStore,
+      isAllowedResource("replicasets") && replicaSetStore,
+      isAllowedResource("jobs") && jobStore,
+      isAllowedResource("cronjobs") && cronJobStore,
+      isAllowedResource("events") && eventStore,
+    ].filter(Boolean);
+
+    const unsubscribeMap = new Map<KubeObjectStore, Function>(
+      stores.map(store => [store, Function])
+    );
 
-    await when(() => this.isUnmounting);
-    unsubscribeList.forEach(dispose => dispose());
+    const loadStores = async () => {
+      this.isLoading = true;
+
+      for (const store of stores) {
+        if (this.isUnmounting) break;
+
+        try {
+          store.reset();
+          await store.loadAll();
+          unsubscribeMap.get(store)(); // unsubscribe previous watcher
```

Comment:
It seems weird to populate a map with a bunch of noops. I think conditionally calling the functions is better.

```suggestion
          unsubscribeMap.get(store)?.(); // unsubscribe previous watcher
```